core/trino-main/src/main/java/io/trino/security/AccessControl.java

```diff
@@ -525,12 +526,20 @@ void checkCanRevokeRoles(SecurityContext context,
      */
     void checkCanExecuteFunction(SecurityContext context, String functionName);
 
+    /**
+     * @throws AccessDeniedException if not allowed
+     * @deprecated Use {@link AccessControl#checkCanExecuteFunction(SecurityContext, QualifiedObjectName, FunctionKind)} instead.
+     * Check if identity is allowed to execute function
+     */
+    @Deprecated
+    void checkCanExecuteFunction(SecurityContext context, QualifiedObjectName functionName);
+
     /**
      * Check if identity is allowed to execute function
      *
      * @throws AccessDeniedException if not allowed
      */
-    void checkCanExecuteFunction(SecurityContext context, QualifiedObjectName functionName);
+    void checkCanExecuteFunction(SecurityContext context, QualifiedObjectName functionName, FunctionKind functionKind);
```

Comment:
I would place function kind before the name.

Reply:
done